zipline/lib/labelarray.py

```diff
@@ -475,9 +476,41 @@ def view(self, dtype=_NotPassed, type=_NotPassed):
             kwargs['type'] = type
         return super(LabelArray, self).view(**kwargs)
 
+    def astype(self,
+               dtype,
+               order='K',
+               casting='unsafe',
+               subok=True,
+               copy=True):
+        if dtype == self.dtype:
+            if copy:
```

Comment:
we can still support this, we if `subok=False` we do: `self.view(type=np.ndarray)`